src/components/layerpublisher/LayerPublisher.js

```diff
@@ -24,43 +43,56 @@ function info(fileName) {
   PubSub.publish(STATUS_INFO, `Loaded ${fileName}.`);
 }
 
+function fetchDataFromDZI(layerType, data) {
+  return fetch(data[layerType].tileSource).then(response => response.text())
+    .then(str => (new window.DOMParser()).parseFromString(str, 'text/xml'))
+    .then((layer) => {
+      const { tileSource } = data[layerType];
+      data[layerType].tileSource = tileSource.substring(0, tileSource.lastIndexOf('/')); // eslint-disable-line no-param-reassign
+      // eslint-disable-next-line no-param-reassign
+      data[layerType].height = layer.getElementsByTagName('Size')[0].attributes.Height.value;
+      // eslint-disable-next-line no-param-reassign
+      data[layerType].width = layer.getElementsByTagName('Size')[0].attributes.Width.value;
+      // eslint-disable-next-line no-param-reassign
+      data[layerType].tileSize = layer.getElementsByTagName('Image')[0].attributes.TileSize.value;
+      return data;
+    });
+}
+
+function fetchImageMetadata(data) {
+  return fetchDataFromDZI('nuclei', data).then(res => fetchDataFromDZI('polyT', res));
+}
+
+function publishLayer(data, type, name, url) {
+  const schema = typeToSchema[type];
```

Comment:
Unless it really needs to be more visible, can we just keep `typeToSchema` in this smaller scope? Is your motivation that it just feels like a configuration that should be moved up and out?

Reply:
That was my motivation, yes, since the information could in theory need be accessed by other functions (and it would make sense that it would be) but if you disagree, I can return it.

Reply:
No strong feelings. Status quo is fine, too.